test_wcd2reshare.py

```diff
@@ -0,0 +1,169 @@
+    )
+
+
+def test_no_rft_fields_of_interest():
+    params = {"rft.popcorn": "sure"}
+    assert wcd2reshare.query_formatter(params) == ""
+
+
+def test_title_priority():
+    params = {
+        "rft.btitle": "btitle",
+        "rft.ctitle": "ctitle",
+        "rft.jtitle": "jtitle",
+        "rft.aulast": "ranch",
+        "rft.title": "title",
+    }
+    assert wcd2reshare.query_formatter(params) == (
+        "join=AND&type0%5B%5D=title&lookfor0%5B%5D=title&"
+        "type0%5B%5D=author&lookfor0%5B%5D=ranch"
+    )
+
+
+def test_btitle_priority():
+    params = {
+        "rft.ctitle": "ctitle",
+        "rft.jtitle": "jtitle",
+        "rft.aulast": "ranch",
+        "rft.btitle": "btitle",
+    }
+    assert wcd2reshare.query_formatter(params) == (
+        "join=AND&type0%5B%5D=title&lookfor0%5B%5D=btitle&"
+        "type0%5B%5D=author&lookfor0%5B%5D=ranch"
+    )
+
+
+def test_ctitle_priority():
+    params = {"rft.jtitle": "jtitle", "rft.aulast": "ranch", "rft.ctitle": "ctitle"}
+    assert wcd2reshare.query_formatter(params) == (
+        "join=AND&type0%5B%5D=title&lookfor0%5B%5D=ctitle&"
+        "type0%5B%5D=author&lookfor0%5B%5D=ranch"
+    )
+
+
+def test_fields_of_no_interest_ignored():
+    params = {"rft.popcorn": "sure", "rft.isbn": "978-3-16-148410-0"}
+    assert wcd2reshare.query_formatter(params) == "type=ISN&lookfor=978-3-16-148410-0"
+
+
+def test_lambda_handler_with_query(baseURL):
+    event = {"queryStringParameters": {"rft.isbn": "978-3-16-148410-0"}}
+    context = {}
+    queryString = "type=ISN&lookfor=978-3-16-148410-0"
+    response = wcd2reshare.lambda_handler(event, context)
+    assert response["statusCode"] == 307
+    assert response["headers"]["Location"] == baseURL + queryString
+
+
+def test_lambda_handler_with_multiple_query(baseURL):
+    event = {
+        "queryStringParameters": {"rft.title": "salad days", "rft.aulast": "ranch"}
+    }
+    context = {}
+    queryString = (
+        "join=AND&type0%5B%5D=title&lookfor0%5B%5D=salad+days&"
+        "type0%5B%5D=author&lookfor0%5B%5D=ranch"
+    )
+    response = wcd2reshare.lambda_handler(event, context)
+    assert response["statusCode"] == 307
+    assert response["headers"]["Location"] == baseURL + queryString
+
+
+def test_lambda_handler_without_query(baseURL):
+    event = {"queryStringParameters": {}}
+    context = {}
+    response = wcd2reshare.lambda_handler(event, context)
+    assert response["statusCode"] == 307
+    assert response["headers"]["Location"] == baseURL
+
+
+def test_lambda_handler_with_garbage_query(baseURL):
+    event = {"queryStringParameters": {"door": "knob"}}
+    context = {}
+    response = wcd2reshare.lambda_handler(event, context)
+    assert response["statusCode"] == 307
+    assert response["headers"]["Location"] == baseURL
+
+
+def test_buildSearchString_no_author():
+    title = "salad days"
+    aulast = None
+    assert wcd2reshare.buildSearchString(title, aulast) == [
```

Comment:
You could also call those values directly rather than as variables

Reply:
title variable is used in the other half of the assert, so I'd have to type it twice if I were to use it directly as an argument.

Reply:
Apologies, should have included that caveat, if it's used twice then definitely make it a variable but it's not always necessary if it's just used once